### PR TITLE
fix: switch notification store to use username instead of userid

### PR DESCRIPTION
**Pull request by iSCJT**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

It has been noticed duplicate users are getting reintroduced into the system and the 'wrong' users are in some cases the only ones getting updated. This was first noticed in the notifications records saved for the user records being different. As a result all uses of the user _id are being switched for userName so that if duplicates are created that it's definitely the user record that has an _id that matches the username that receives updates.  This pr:

- Switches use of user._id to user._userName in the notifications store.
- Fixes 'notifications' typos

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Comment by thisislawatts** [comment]
@iSCJT could you explain why this change is being made?
> Switches use of user._id to user._userName

**Comment by iSCJT** [comment]
> @iSCJT could you explain why this change is being made?
> 
> > Switches use of user._id to user._userName

I've updated the pr description @thisislawatts

File: src/stores/User/notifications.store.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ export class UserNotificationsStore extends ModuleStore {
           _created: new Date().toISOString(),
           triggeredBy: {
             displayName: triggeredBy.displayName,
-            userId: triggeredBy._id,
+            userId: triggeredBy.userName,
```
Comment (evakill) [inline src/stores/User/notifications.store.ts:75]: would it be a big diff to change this to be called userName for clarity?

Comment (iSCJT) [reply src/stores/User/notifications.store.ts:75]: had a think and would definitely be clearer although we'd need to run some kind of data migration for existing records which I'm not sure is worth the time to code right now.  I'd prefer to hold off and then get a proper db with a relational notifications table 🤓

Comment (evakill) [reply src/stores/User/notifications.store.ts:75]: fair! in that case could you add a comment / todo to this code explaining the reasoning?

**Review by evakill — APPROVED** [review]

**Pull request merged**